core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestParser.java

```diff
@@ -98,18 +98,10 @@ public PhraseSuggestParser(PhraseSuggester suggester) {
                     }
                 }
             } else if (token == Token.START_ARRAY) {
-                if ("direct_generator".equals(fieldName) || "directGenerator".equals(fieldName)) {
+                if (parseFieldMatcher.match(fieldName, org.elasticsearch.search.suggest.phrase.PhraseSuggestionBuilder.DirectCandidateGenerator.DIRECT_GENERATOR_FIELD)) {
```

Comment:
Can we make this either `PhraseSuggestionBuilder.DirectCandidateGenerator` or `DirectCandidateGenerator`?